### PR TITLE
2457 fix code climate test reported for mcbe

### PR DESCRIPTION
**Pull request by rizzkhan1**
### Context
Integration of CodeClimate with MCBE Azure Pipelines
 
### Changes proposed in this pull request
Update Pipeline variables
Update dockerfile and docker-compose for variable pass thru
Add CodeClimate setup task
Add CodeClimate reporter task

### Guidance to review
CodeClimate badge hasn't updated yet but test_reporter is receiving the status sent by the CI pipeline 
https://codeclimate.com/repos/5cd45a66af3d500177000666/settings/test_reporter

###NOTES: Microsoft documentation is invalid:
`AGENT_JOBSTATUS` doesn't work `AGENT.JOBSTATUS`  is a valid variable for `TRAVIS_TEST_RESULT` replacement
https://docs.microsoft.com/en-us/azure/devops/pipelines/migrate/from-travis?view=azure-devops

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -30,11 +31,13 @@ steps:
     $DOCKER_OVERRIDE up --no-build -d
     $DOCKER_OVERRIDE exec -T web /bin/sh -c "./wait-for-command.sh -c 'nc -z db 5432' -s 0 -t 20"
     $DOCKER_OVERRIDE exec -T web /bin/sh -c "bundle exec rails db:setup"
+    $DOCKER_OVERRIDE exec -T web /bin/sh -c "bundle exec rake cc:setup"
```
Comment (defong) [inline azure-pipelines.yml]: can we move the curl install to here as its not really needed for the image

Comment (dankmitchell) [reply azure-pipelines.yml]: Good shout

Comment (rizzkhan1) [reply azure-pipelines.yml]: Moved `apk add curl` out of docker image and into the pipeline as suggested by Sau

Comment (defong) [reply azure-pipelines.yml]: beautiful

File: azure-pipelines.yml [hunk 2]
```diff
@@ -49,6 +53,15 @@ steps:
     DOCKER_OVERRIDE: $(dockerOverride)
     dockerHubUsername: $(dockerHubUsername)
     dockerHubImageName: $(imageName)
+- script: |
+    $DOCKER_OVERRIDE exec -T web /bin/sh -c "bundle exec rake cc:report"
```
Comment (defong) [inline azure-pipelines.yml]: do you need to add it here as well?

Comment (rizzkhan1) [reply azure-pipelines.yml]: This is a report task `cc:report`, earlier one was a setup task `cc:setup`. Reporter sends a success or fail message to Code Climate

**Pull request merged**